dotnet/src/dotnetframework/GxClasses/Core/GXUtilsCommon.cs

```diff
@@ -2466,14 +2466,41 @@ public DateTime YMDHMSMToT(int year, int month, int day, int hour, int min, int
 		{
 			return YMDHMSMToT(year, month, day, hour, min, sec, mil, true);
 		}
+		private bool IsValidDate(int year, int month, int day)
+		{
+			if (year < DateTime.MinValue.Year || year > DateTime.MaxValue.Year)
+				return false;
+
+			if (month < 1 || month > 12)
+				return false;
 
+			return day > 0 && day <= DateTime.DaysInMonth(year, month);
+		}
+		private bool IsValidTime(int hour, int min, int sec, int mil)
+		{
+			return hour >= 0 && hour <24 && min >= 0 && min < 60 && sec >= 0 && sec < 60 && mil >= 0 && mil < 1000;
+		}
 		public DateTime YMDHMSMToT(int year, int month, int day, int hour, int min, int sec, int mil, bool centuryConversion)
 		{
 			try
 			{
+
 				int year1 = year;
 				if (centuryConversion)
 					year1 = GetYear(year);
+				if (!IsValidDate(year, month, day))
```

Comment:
I think that the first parameter should be "year1" instead of year, because "year " may not be a full 4 digits year.